web/ui/src/app/lineage/graph/graph-builder.ts

```diff
@@ -34,13 +37,74 @@ export function lineageToGraph(lineage: IDataLineage,
         visibleOperations: IOperation[] = operationsByVisibility.true,
         hiddenOperations: IOperation[] = operationsByVisibility.false,
         hiddenOpIds: string[] = _.map(hiddenOperations, "mainProps.id"),
-        visibleNodes = visibleOperations.map(op => new RegularVisNode(op)),
+        visibleNodes = visibleOperations.map(op => new RegularVisNode(op, getLabel(op, expressionRenderService))),
         visibleEdges = createVisibleEdges(lineage, hiddenOpIds)
 
     return new VisModel(
         new vis.DataSet<VisNode>(visibleNodes),
         new vis.DataSet<VisEdge>(visibleEdges))
 }
+/**
+ * Get label from an operation
+ * 
+ * @param operation the operation where the label should be retrieved
+ * @param expressionRenderService service to render expressions
+ * 
+ * @returns the label from the operation
+ */
+export function getLabel(operation : (IOperation), expressionRenderService : ExpressionRenderService){
+    let label = operation.mainProps.name + "\n"; 
+    switch(typeOfOperation(operation)){
+        case "Aggregate":
+            let aggregations = (<any> operation).aggregations;
+            label += Object.keys(aggregations).join(",");break;
+        case "Join":
+            let joins = (<IJoin> operation).condition;
```

Comment:
Try to find more meaningful names for variables. The names like `joins` or `filters` don't make it obvious that they contain a value of a singular expression type.

Reply:
Renammed